tiledb/sm/array_schema/enumeration.cc

```diff
@@ -48,6 +48,7 @@ class EnumerationException : public StatusException {
 };
 
 Enumeration::Enumeration(
+    shared_ptr<MemoryTracker> memory_tracker,
```

Comment:
Same as for the ArraySchema PR. Lets move all memory trackers to the last argument whenever possible to maintain consistency. Same as the ArraySchema PR I can handle this one as well if you like.

Reply:
I can also handle this one!